### PR TITLE
✨ Rewards Modal

### LINKED ISSUE #1111: Rewards modal

### PR DESCRIPTION
**Pull request by JuampiRombola**
closes #1111

### CODE REVIEW

**Review by sebipap — COMMENTED** [review]
lgtm! added minor comments

File: components/common/modal/Loading/index.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  const isLoading = useMemo(() => tx.status === 'processing' || tx.status === 'loading', [tx]);
+  const isSuccess = useMemo(() => tx.status === 'success', [tx]);
+  const isError = useMemo(() => tx.status === 'error', [tx]);
+  const { tx: txLink } = useEtherscanLink();
+
+  return (
+    <Box display="flex" minWidth="300px">
+      <Box display="flex" flexDirection="column" mx="auto" alignItems="center" py={{ xs: '30px', sm: '0' }} gap="32px">
+        {isLoading && <CircularProgress size={100} thickness={1.5} />}
```
[the remark this answers is not in the document]
Comment (jgalat) [reply components/common/modal/Loading/index.tsx:33]: IMO this is more readable than having `if...elseif...else`

File: components/RewardsModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+                      }
+                    />
+                  ))}
+                  <Collapse in={showInput}>
+                    <Box mt={2.5}>
+                      <TextField
+                        variant="outlined"
+                        fullWidth
+                        size="small"
+                        placeholder={t('Enter address') || ''}
```
[the remark this answers is not in the document]
Comment (jgalat) [reply components/RewardsModal/index.tsx:243]: Good catch. I think `t` use to return `string | null`, i see several places where adding a `t` forced us to change the prop type to also consider `null`. Checking it now seems to only return `string`, we should change this and all other appearances of `| null`.

[the remark this answers is not in the document]
Comment (jgalat) [reply components/RewardsModal/index.tsx]: addressed in 214241f26b4f5040163c1d2895edd8af88d35a17

**Review by jgalat — APPROVED** [review]

**Review by franm91 — APPROVED** [review]

File: components/RewardsModal/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (sebipap) [reply components/RewardsModal/index.tsx]: there's an [eslint rule](https://typescript-eslint.io/rules/no-unnecessary-condition/) to avoid this

**Pull request merged**